lib/runtime-core/src/error.rs

```diff
@@ -100,37 +100,37 @@ pub enum LinkError {
         name: String,
     },
     /// The memory descriptor provided does not match the expected descriptor.
-    IncorrectMemoryDescriptor {
+    IncorrectMemoryType {
```

Comment:
Something I want reviewers to note about this PR is that it introduces breaking changes in `LinkError`, the error variants are renamed here:  I don't see a good way to make this change without breaking these.  Well, unless we're fine with the errors referring to them as `descriptors`